extensions/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java

```diff
@@ -406,6 +406,7 @@ private static void generateDefaultConstructors(BuildProducer<BytecodeTransforme
 
         final Set<String> allowedAnnotationPrefixes = new HashSet<>(1 + additionalJaxRsResourceDefiningAnnotations.size());
         allowedAnnotationPrefixes.add(packageName(ResteasyDotNames.PATH));
+        allowedAnnotationPrefixes.add("kotlin"); // make sure the annotation that the Kotlin compiler adds don't interfere with creating a default constructor
```

Comment:
Do we need to do the same for `scala`?

Reply:
Good question! 

I have :zero: knowledge of Scala TBH so I really don't know what the Scala compiler produces. I would say that if you or anyone else wants to check it out, then you can follow up with a new PR if needed :)